src/Val.h

```diff
@@ -965,7 +965,27 @@ class RecordVal final : public Val, public notifier::Modifiable {
 	void Assign(int field, std::nullptr_t)
 		{ Assign(field, IntrusivePtr<Val>{}); }
 
-	Val* Lookup(int field) const;	// Does not Ref() value.
+	[[deprecated("Remove in v4.1.  Use GetField().")]]
+	Val* Lookup(int field) const	// Does not Ref() value.
+		{ return (*AsRecord())[field].get(); }
```

Comment:
`operator[]` does not do bounds checking, but `at()` does (it throws an exception on invalid index) and I prefer to avoid adding any of the overhead associated with that.